injectors/index.js

```diff
@@ -27,34 +27,44 @@ try {
   }
 }
 
-const VALID_PLATFORMS = [ 'stable', 'ptb', 'canary', 'dev', 'development' ];
-
-let platform = (
-  process.argv.find(x => VALID_PLATFORMS.includes(x.toLowerCase())) || 'stable'
-).toLowerCase();
-
-if (!VALID_PLATFORMS.includes(platform)) {
-	console.log(`${AnsiEscapes.RED}Platform you specified isn't valid, please specify a valid one.${AnsiEscapes.RESET}\n\nList of valid platforms:\n${AnsiEscapes.GREEN}${VALID_PLATFORMS.map(x => `${x}`).join('\n')}${AnsiEscapes.RESET}`);
-	process.exit(process.argv.includes('--no-exit-codes') ? 0 : 1);
-}
+let platform = process.argv[4]?.toLowerCase();
 
 (async () => {
-  if (!process.argv.find(x => VALID_PLATFORMS.includes(x.toLowerCase()))) {
-    console.log(`${AnsiEscapes.YELLOW}No valid platform specified, defaulting to "${platform}".${AnsiEscapes.RESET}\n`);
-  } else if (platform === 'development') {
-    platform = 'dev';
+  if (platform) {
+    const exists = main.checkPlatform(platform);
+    if (!exists) {
+      console.log(`${AnsiEscapes.RED}Platform you specified isn't valid, please specify a valid one.${AnsiEscapes.RESET}\n\nList of valid platforms:\n${AnsiEscapes.GREEN}${main.VALID_PLATFORMS.map(x => `${x}`).join('\n')}${AnsiEscapes.RESET}`);
+      process.exit(process.argv.includes('--no-exit-codes') ? 0 : 1);
+    } else if (platform === 'development') {
+      platform = 'dev';
+    }
   } else {
-    platform = platform.toLowerCase();
+    for (const current of main.VALID_PLATFORMS) {
+      if (current !== 'development') {
+        try {
+          const installed = await main.checkInstalled(platformModule, current);
+          if (installed) {
+            console.log(`${AnsiEscapes.GREEN}${current} is installed, attempting to ${process.argv[2] === 'uninject' ? 'unplug' : 'plug'}.${AnsiEscapes.RESET}`);
+            platform = current;
+            break;
+          } else {
+            console.log(`${AnsiEscapes.RED}${current} is not installed, skipping.${AnsiEscapes.RESET}`);
+          }
+        } catch (e) {
+          console.log(`${AnsiEscapes.RED}${current} is not installed, skipping.${AnsiEscapes.RESET}`);
+        }
+      }
+    }
   }
 
```

Comment:
If platform doesn't exist here, say "Could not find any installations of Discord" and exit